### PR TITLE
Feature/1260 fallback retry timeout option

### LINKED ISSUE #1260: Missing properties in ARTClientOptions
Captured in following issue 
https://github.com/ably/ably-flutter/issues/14

There are some properties missing  in ARTClientOptions. I checked ones that were not relevant anymore. The remaining properties
* `useTokenAuth`
* `realtimeRequestTimeout`
* `fallbackRetryTimeout`
* `asyncHttpThreadpoolSize`

For details of the properties please see linked issue.

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-1451) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by maratal**
Closes #1260

### CODE REVIEW

**Review by lukasz-szyszkowski — APPROVED** [review]

File: Spec/Tests/RestClientTests.swift [hunk 1]
```diff
@@ -1533,46 +1533,32 @@ class RestClientTests: XCTestCase {
...
     }
 
-    func beforeEach__RestClient__Host_Fallback__should_store_successful_fallback_host_as_default_host__should_restore_default_primary_host_after_fallbackRetryTimeout_expired() {
-        ARTDefault.setFallbackRetryTimeout(1.0)
-    }
-
     func test__080__RestClient__Host_Fallback__should_store_successful_fallback_host_as_default_host__should_restore_default_primary_host_after_fallbackRetryTimeout_expired___hostUnreachable() {
-        beforeEach__RestClient__Host_Fallback__should_store_successful_fallback_host_as_default_host__should_restore_default_primary_host_after_fallbackRetryTimeout_expired()
-
+
```
Comment (lawrence-forooghian) [inline Spec/Tests/RestClientTests.swift:1537]: (minor) I don't think we need this empty line and the ones in the subsequent methods?

File: Spec/Tests/RestClientTests.swift [hunk 2]
```diff
@@ -1533,46 +1533,32 @@ class RestClientTests: XCTestCase {
         testStoresSuccessfulFallbackHostAsDefaultHost(.hostInternalError(code: 501))
     }
 
-    func beforeEach__RestClient__Host_Fallback__should_store_successful_fallback_host_as_default_host__should_restore_default_primary_host_after_fallbackRetryTimeout_expired() {
-        ARTDefault.setFallbackRetryTimeout(1.0)
-    }
-
     func test__080__RestClient__Host_Fallback__should_store_successful_fallback_host_as_default_host__should_restore_default_primary_host_after_fallbackRetryTimeout_expired___hostUnreachable() {
-        beforeEach__RestClient__Host_Fallback__should_store_successful_fallback_host_as_default_host__should_restore_default_primary_host_after_fallbackRetryTimeout_expired()
```
Comment (lawrence-forooghian) [inline Spec/Tests/RestClientTests.swift:1541]: If you're going to remove this `beforeEach`, don't its contents need to go inside `testRestoresDefaultPrimaryHostAfterTimeoutExpires`, similarly to what you've done for `testUsesAnotherFallbackHost`?

File: Source/ARTClientOptions.m [hunk 3]
```diff
@@ -120,7 +121,7 @@ - (id)copyWithZone:(NSZone *)zone {
     options.httpMaxRetryCount = self.httpMaxRetryCount;
     options.httpMaxRetryDuration = self.httpMaxRetryDuration;
     options.httpOpenTimeout = self.httpOpenTimeout;
-    options.httpRequestTimeout = self.httpRequestTimeout;
```
[the remark this answers is not in the document]
Comment (maratal) [reply Source/ARTClientOptions.m:123]: Yes, it's duplicated at line 132

File: Spec/Tests/RestClientTests.swift (hunk 1 above)
Comment (maratal) [reply Spec/Tests/RestClientTests.swift:1537]: 82868c34

File: Spec/Tests/RestClientTests.swift (hunk 2 above)
Comment (maratal) [reply Spec/Tests/RestClientTests.swift:1541]: I did, look at line 83 - `options.fallbackRetryTimeout = 1`

File: Source/ARTClientOptions.h [hunk 4]
```diff
@@ -67,6 +67,11 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (readwrite, assign, nonatomic) NSTimeInterval httpRequestTimeout;
 
+/**
+ The period in seconds before HTTP requests are retried against the default endpoint
```
[the remark this answers is not in the document]
Comment (maratal) [reply Source/ARTClientOptions.h]: b9b426b69c6e0d5741e235a083284ede6fbaa964

File: Spec/Tests/RestClientTests.swift (hunk 1 above)
Comment (maratal) [reply Spec/Tests/RestClientTests.swift]: 82868c34c35070fe4c2ff8881255e4a3703d2f4e

**Review by QuintinWillison — APPROVED** [review]

File: Source/ARTClientOptions.m (hunk 3 above)
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Source/ARTClientOptions.m:123]: Ah, missed that, thanks!

**Review by lawrence-forooghian — APPROVED** [review]

**Pull request merged**